relocation R_X86_64_32S against undefined symbol `PREFETCHSIZE' can not be used when making a shared object

Basically it's the same as 5376, but on a different processor:

cpuid:
[..]
   (multi-processing method) = Intel leaf 0xb
   (APIC widths synth): CORE_width=6 SMT_width=1
   (APIC synth): PKG_ID=0 CORE_ID=3 SMT_ID=0
   (uarch synth) = Intel Tremont, 10nm
   (synth) = Intel Celeron N4500 / N5100 (Jasper Lake A0/A1) [Tremont], 10nm

cat /proc/cpuinfo:
[..]
processor       : 3
vendor_id       : GenuineIntel
cpu family      : 6
model           : 156
model name      : Intel(R) Celeron(R) N5105 @ 2.00GHz
stepping        : 0
microcode       : 0x2400001e
cpu MHz         : 2879.429
cache size      : 4096 KB
physical id     : 0
siblings        : 4
core id         : 3
cpu cores       : 4
apicid          : 6
initial apicid  : 6
fpu             : yes
fpu_exception   : yes
cpuid level     : 27
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg cx16 xtpr pdcm sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave rdrand lahf_lm 3dnowprefetch cpuid_fault epb cat_l2 cdp_l2 ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust smep erms rdt_a rdseed smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req vnmi umip waitpkg gfni rdpid movdiri movdir64b md_clear flush_l1d arch_capabilities
vmx flags       : vnmi preemption_timer posted_intr invvpid ept_x_only ept_ad flexpriority apicv tsc_offset vtpr mtf vapic ept vpid unrestricted_guest vapic_reg vid ple shadow_vmcs ept_violation_ve ept_mode_based_exec tsc_scaling usr_wait_pause
bugs            : spectre_v1 spectre_v2 spec_store_bypass swapgs srbds mmio_stale_data rfds bhi spectre_v2_user old_microcode
bogomips        : 3993.60
clflush size    : 64
cache_alignment : 64
address sizes   : 39 bits physical, 48 bits virtual
power management:

Result compiling:
./gensymbol linktest  x86_64 "_" 1 0 0 0 0 0 "" "" 1 0 0 1 1 1 1 > linktest.c
x86_64-pc-linux-gnu-gcc -march=native -O2 -pipe   -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -DEXPRECISION -m128bit-long-double -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=64 -DMAX_PARALLEL_NUMBER=8 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.31\" -msse3 -mssse3 -msse4.1 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I.   -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -DEXPRECISION -m128bit-long-double -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=64 -DMAX_PARALLEL_NUMBER=8 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.31\" -msse3 -mssse3 -msse4.1 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I.. -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -shared -o ../libopenblas_prescottp-r0.3.31.so \
-Wl,--whole-archive ../libopenblas_prescottp-r0.3.31.a -Wl,--no-whole-archive \
-Wl,-soname,libopenblas.so.0 -lm -lpthread -lgfortran -lm -lpthread -lgfortran
/usr/x86_64-pc-linux-gnu/binutils-bin/2.46.0/ld: ../libopenblas_prescottp-r0.3.31.a(ssymv_U.o): relocation R_X86_64_32S against undefined symbol `PREFETCHSIZE' can not be used when making a shared object; recompile with -fPIC
/usr/x86_64-pc-linux-gnu/binutils-bin/2.46.0/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:210: ../libopenblas_prescottp-r0.3.31.so] Error 1
make[1]: Leaving directory '/var/tmp/portage/sci-libs/openblas-0.3.31/work/OpenBLAS-0.3.31/exports'
make: *** [Makefile:149: shared] Error 2

--- a/cpuid_x86.c
+++ b/cpuid_x86.c
@@ -1588,6 +1588,8 @@ int get_cpuname(void)
           if (support_avx2()) return CPUTYPE_HASWELL;
           if (support_avx()) return CPUTYPE_SANDYBRIDGE;
           else return CPUTYPE_NEHALEM;
+	case 12: // Jasper Lake Celeron N5105
+	  return CPUTYPE_NEHALEM;
         case 13: // Ice Lake NNPI
           if (support_avx512()) return CPUTYPE_SKYLAKEX;
           if (support_avx2()) return CPUTYPE_HASWELL;
@@ -2302,6 +2304,8 @@ int get_coretype(void)
           if (support_avx()) return CORE_SANDYBRIDGE;
           else return CORE_NEHALEM;
         }
+	if (model == 12) // Jasper Lake
+	  return CORE_NEHALEM;
         if (model == 13) { // Ice Lake NNPI
           if (support_avx512()) return CORE_SKYLAKEX;
           if (support_avx2()) return CORE_HASWELL;
@@ -2400,6 +2404,7 @@ int get_coretype(void)
       if (support_avx512()) return CORE_SKYLAKEX;
       if (support_avx2()) return CORE_HASWELL;
       if (support_avx()) return CORE_SANDYBRIDGE;
+      return CORE_NEHALEM;
     }
   }
 

